RecoTracker/FinalTrackSelectors/plugins/TrackTfClassifier.cc

```diff
@@ -34,72 +36,115 @@ namespace {
-      input1.matrix<float>()(0, 13) = trk.dz(beamSpot.position());
-      input1.matrix<float>()(0, 14) = trk.dxyError();
-      input1.matrix<float>()(0, 15) = trk.dzError();
-      input1.matrix<float>()(0, 16) = trk.normalizedChi2();
-      input1.matrix<float>()(0, 17) = trk.eta();
-      input1.matrix<float>()(0, 18) = trk.phi();
-      input1.matrix<float>()(0, 19) = trk.etaError();
-      input1.matrix<float>()(0, 20) = trk.phiError();
-      input1.matrix<float>()(0, 21) = trk.hitPattern().numberOfValidPixelHits();
-      input1.matrix<float>()(0, 22) = trk.hitPattern().numberOfValidStripHits();
-      input1.matrix<float>()(0, 23) = trk.ndof();
-      input1.matrix<float>()(0, 24) = trk.hitPattern().numberOfLostTrackerHits(reco::HitPattern::MISSING_INNER_HITS);
-      input1.matrix<float>()(0, 25) = trk.hitPattern().numberOfLostTrackerHits(reco::HitPattern::MISSING_OUTER_HITS);
-      input1.matrix<float>()(0, 26) =
-          trk.hitPattern().trackerLayersTotallyOffOrBad(reco::HitPattern::MISSING_INNER_HITS);
-      input1.matrix<float>()(0, 27) =
-          trk.hitPattern().trackerLayersTotallyOffOrBad(reco::HitPattern::MISSING_OUTER_HITS);
-      input1.matrix<float>()(0, 28) = trk.hitPattern().trackerLayersWithoutMeasurement(reco::HitPattern::TRACK_HITS);
-
-      //Original algo as its own input, it will enter the graph so that it gets one-hot encoded, as is the preferred
-      //format for categorical inputs, where the labels do not have any metric amongst them
-      input2.matrix<float>()(0, 0) = trk.originalAlgo();
-
-      //The names for the input tensors get locked when freezing the trained tensorflow model. The NamedTensors must
-      //match those names
-      tensorflow::NamedTensorList inputs;
-      inputs.resize(2);
-      inputs[0] = tensorflow::NamedTensor("x", input1);
-      inputs[1] = tensorflow::NamedTensor("y", input2);
-      std::vector<tensorflow::Tensor> outputs;
-
-      //evaluate the input
-      tensorflow::run(const_cast<tensorflow::Session*>(session_), inputs, {"Identity"}, &outputs);
-      //scale output to be [-1, 1] due to convention
-      float output = 2.0 * outputs[0].matrix<float>()(0, 0) - 1.0;
+    std::vector<float> operator()(reco::TrackCollection const& tracks,
+                                  reco::BeamSpot const& beamSpot,
+                                  reco::VertexCollection const& vertices) const {
+      int size_in = (int)tracks.size();
+      int nbatches = size_in / bsize_;
+
+      std::vector<float> output;
+      output.resize(size_in);
+
+      tensorflow::Tensor input1(tensorflow::DT_FLOAT, {bsize_, 29});
+      tensorflow::Tensor input2(tensorflow::DT_FLOAT, {bsize_, 1});
+
+      for (auto nb = 0; nb < nbatches + 1; nb++) {
+        for (auto nt = 0; nt < bsize_; nt++) {
+          int itrack = nt + bsize_ * nb;
+          if (itrack >= size_in)
+            continue;
+          const auto& trk = tracks.at(itrack);
```

Comment:
Usually we ask for the no-exception version, especially in performance-critical code.
```suggestion
          const auto& trk = tracks[itrack];
```

Reply:
I should have done this already as in 37878. sorry for the double review.